### ISSUE TITLE
Release 0.12.8: Running great_expectations init throws this error: "ModuleNotFoundError: No module named 's3fs'"

### ISSUE THREAD

**Issue by eugmandel** [fivetran/great_expectations#2053]
**Describe the bug**
Release 0.12.8: Running great_expectations init throws this error: "ModuleNotFoundError: No module named 's3fs'"

**To Reproduce**
1. pip install the 0.12.8 release of Great Expectations in a clean Python environment
2. Run `great_expectations init` CLI command
3. See error

**Expected behavior**
Since s3fs is an optional module, no error should be raised unless you are using a class that requires it and it is not installed.

**Environment (please complete the following information):**
 - OS: any
 - GE Version: 0.12.8

**Comment by eugmandel** [comment]
This bug was introduced in 0.12.8 in the new S3SubdirReaderBatchKwargsGenerator class.

Will fix the code to make s3fs optional. 

The pattern we are using in this case is this:

instead of 

`import s3fs`

```
try:
    import s3fs
except ImportError:
    s3fs = None

```

and then 

```
if not s3fs:
    raise ClassInstantiationError("ModuleNotFoundError: No module named 's3fs'")

```

before using the module.

**Comment by anthonyburdi** [comment]
Closed via #2054
